tests/test_linalg.py

```diff
@@ -139,3 +140,67 @@ def test_dgemm_input_handling():
         dgemm(alpha=alpha,
               view_a=view_a,
               view_b=view_b)
+
+
+@pytest.mark.parametrize("alpha, a, b, expected", [
+    (1.0,
+     np.ones((4, 4)),
+     np.ones((4, 4)),
+     np.full((4, 4), 4),
+    ),
+    (1.0,
+    np.eye(4, 4),
+    np.array([[0, 6, 5, 0],
+             [9, 2, 2, 1],
+             [3, 1, 3, 8],
+             [4, 9, 4, 2]], dtype=float),
+    np.array([[0, 6, 5, 0],
+             [9, 2, 2, 1],
+             [3, 1, 3, 8],
+             [4, 9, 4, 2]], dtype=float),
+    ),
+    (1.0,
+    np.ones((4, 4)),
+    np.array([[0, 6, 5, 0],
+             [9, 2, 2, 1],
+             [3, 1, 3, 8],
+             [4, 9, 4, 2]], dtype=float),
+    np.array([[16., 18., 14., 11.],
+              [16., 18., 14., 11.],
+              [16., 18., 14., 11.],
+              [16., 18., 14., 11.]], dtype=float)
+
+    ),
+    ])
+def test_dgemm_tiled(alpha, a, b, expected):
+    # expected values hardcoded from SciPy output
+    actual_c = dgemm(alpha=alpha,
+                     view_a=pk.from_numpy(a),
+                     view_b=pk.from_numpy(b),
+                     beta=0.0,
+                     view_c=None,
+                     tile_width=2)
+    assert_allclose(actual_c, expected)
+
+
+@pytest.mark.parametrize("input_width, tile_width", [
+    (4, 2),
+    #(8, 2),
```

Comment:
The 2 x 2 tiled matrix multiplication tests are now passing for me locally at least, with SciPy (OpenBLAS) `dgemm` as a reference, when 4x4 matrices are used for `a`, and `b`.

Other matrix and/or tile sizes can cause substantial issues, but perhaps I've carried this far enough forward for an initial review of the tiling algorithm and suggestions for generalization to varied matrix and tile sizes would be helpful (within the confines of i.e., powers of 2 at least).

Also, I believe we may need to add checks to prevent certain types of segfaults/errors, for example requesting team (thread) sizes/league (block) sizes or shared memory sizes that are not hardware-compatible...

Reply:
@JBludau The segfault in CI vs. passing locally is possibly fairly low-level hierarchical parallelism sutff, but having this lower-level stuff carefully caught with a thoughtful error message would be most helpful for debugging.